### PR TITLE
add prod at numpy_frontend 

### PR DESCRIPTION
**Pull request by AmeenUrRehman**
Issue no. #5713

### CODE REVIEW

**Comment by AmeenUrRehman** [comment]
@juliagsy plz check this could we ready to merge. Thank You !!!

**Comment by AmeenUrRehman** [comment]
@JamieLine could we be ready to merge plz check !!!

**Comment by AmeenUrRehman** [comment]
@juliagsy hey, could we merge this plz check !!!

File: ivy/functional/frontends/numpy/ndarray/ndarray.py [hunk 1]
```diff
@@ -57,3 +57,8 @@ def any(self, axis=None, out=None, keepdims=False, *, where=True):
 
     def argsort(self, *, axis=-1, kind=None, order=None):
         return np_frontend.argsort(self.data, axis, kind, order)
+
+    def prod(self , axis=None, dtype=None, out=None, keepdims=False, initial=1, where=True):
```
Comment (juliagsy) [inline ivy/functional/frontends/numpy/ndarray/ndarray.py:61]: (for lint fix) please format the parameters such that `param1, param2,...` (comma and spaces formatting)

File: ivy/functional/frontends/numpy/ndarray/ndarray.py [hunk 2]
```diff
@@ -57,3 +57,8 @@ def any(self, axis=None, out=None, keepdims=False, *, where=True):
...
+        return np_frontend.prod(self.data , axis , dtype , out, keepdims , initial , where= where)
```
Comment (juliagsy) [inline ivy/functional/frontends/numpy/ndarray/ndarray.py:62]: all parameters beginning from axis are keyword args (i.e., `axis=axis, dtype=dtype,...`)

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey there! Great work so far! Just some small changes on formatting and keyword args, please also ensure that tests are passing!

As usual, please request my review again when you're done! Thanks!

**Comment by AmeenUrRehman** [comment]
@juliagsy hey, thanks I have fixed the issues could you plz check this !!!

File: ivy/functional/frontends/numpy/ndarray/ndarray.py (hunk 2 above)
Comment (AmeenUrRehman) [reply ivy/functional/frontends/numpy/ndarray/ndarray.py]: @juliagsy done

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey! Thanks for the changes! Great work so far! I've just realised, a test is needed for this instance method in the `ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py` file

If the tests are passing, you'll be good to go soon!

**Pull request closed without merging**